src/block_service/hdfs/hdfs_service.cpp

```diff
@@ -0,0 +1,455 @@
+#include "hdfs_service.h"
+
+#include <algorithm>
+#include <fstream>
+
+#include <dsn/dist/fmt_logging.h>
+#include <dsn/tool-api/async_calls.h>
+#include <dsn/tool-api/task.h>
+#include <dsn/tool-api/task_tracker.h>
+#include <dsn/utility/error_code.h>
+#include <dsn/utility/filesystem.h>
+#include <dsn/utility/flags.h>
+#include <dsn/utility/safe_strerror_posix.h>
+#include <dsn/utility/utils.h>
+
+namespace dsn {
+namespace dist {
+namespace block_service {
+
+DEFINE_THREAD_POOL_CODE(THREAD_POOL_HDFS_SERVICE)
+DEFINE_TASK_CODE(LPC_HDFS_SERVICE_CALL, TASK_PRIORITY_COMMON, THREAD_POOL_HDFS_SERVICE)
+
+DSN_DEFINE_uint64("replication",
+                  hdfs_read_batch_size_bytes,
+                  64 << 20,
+                  "hdfs read batch size, the default value is 64MB");
+
+DSN_DEFINE_uint64("replication",
+                  hdfs_write_batch_size_bytes,
+                  64 << 20,
+                  "hdfs write batch size, the default value is 64MB");
```

Comment:
I recommend adding config changes above in pr description and add `config-change` tag for this pr.